app/src/ui/diff/submodule-diff.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react'
+import { parseRepositoryIdentifier } from '../../lib/remote-parsing'
+import { shortenSHA } from '../../models/commit'
+import { ISubmoduleDiff } from '../../models/diff'
+import { LinkButton } from '../lib/link-button'
+import { Ref } from '../lib/ref'
+import { TooltippedCommitSHA } from '../lib/tooltipped-commit-sha'
+import { Octicon } from '../octicons'
+import * as OcticonSymbol from '../octicons/octicons.generated'
+import { SuggestedAction } from '../suggested-actions'
+
+type SubmoduleItemIcon =
+  | {
+      readonly octicon: typeof OcticonSymbol.info
+      readonly className: 'info-icon'
+    }
+  | {
+      readonly octicon: typeof OcticonSymbol.diffModified
+      readonly className: 'modified-icon'
+    }
+  | {
+      readonly octicon: typeof OcticonSymbol.fileDiff
+      readonly className: 'untracked-icon'
+    }
+
+interface ISubmoduleDiffProps {
+  readonly onOpenSubmodule?: (fullPath: string) => void
+  readonly diff: ISubmoduleDiff
+}
+
+export class SubmoduleDiff extends React.Component<ISubmoduleDiffProps> {
+  public constructor(props: ISubmoduleDiffProps) {
+    super(props)
+  }
+
+  public render() {
+    return (
+      <div className="changes-interstitial submodule-diff">
+        <div className="content">
+          <div className="interstitial-header">
+            <div className="text">
+              <h1>Submodule changes</h1>
+            </div>
+          </div>
+          {this.renderSubmoduleInfo()}
+          {this.renderCommitChangeInfo()}
+          {this.renderSubmodulesChangesInfo()}
+          {this.renderOpenSubmoduleAction()}
+        </div>
+      </div>
+    )
+  }
+
+  private renderSubmoduleInfo() {
+    // TODO: only for GH submodules?
```

Comment:
Just remembered - I meant to ask if this is needed - does it work just fine for non GH submodules?

Reply:
I only tested this with GH and bitbucket repos and it worked fine, but maybe it doesn't with other hosting services? 🤔